src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NonGCStaticsNode.cs

```diff
@@ -18,19 +18,12 @@ namespace ILCompiler.DependencyAnalysis
     internal class NonGCStaticsNode : ObjectNode, ISymbolNode
     {
         private MetadataType _type;
-        private ISymbolNode _classConstructorContext;
+        private bool _hasLazyClassConstructor = false;
```

Comment:
There's little point in caching this. It's used in exactly one spot and it doesn't save us much.